### PR TITLE
Normalize workspace path inputs

### LINKED ISSUE #74: 경로 입력 정규화: file:/// URI·따옴표·슬래시 변형 수용
#56에서 MCP 클라이언트가 경로를 `file:///C:/...` URI 형태로 넘겨 실패하는 사례가 보고되었습니다. LLM 클라이언트는 경로를 다양한 변형(file URI, 따옴표 감쌈, 혼합 슬래시)으로 보낼 수 있으므로, 워크스페이스 검증 전에 이런 변형을 정규화하면 실사용 성공률이 올라갑니다.

**제안**: `resolve_path` 진입 시 (1) `file://` URI → 로컬 경로 변환, (2) 앞뒤 따옴표 제거, (3) Windows 역슬래시/슬래시 정규화. 워크스페이스 경계 검증 자체는 그대로 유지합니다(보안 계약 불변).

**관련 코드**: `src/hwpx_mcp_server/utils/helpers.py`의 `resolve_path`, `src/hwpx_mcp_server/workspace.py`. 테스트는 `tests/test_workspace.py` 참고.

### PR DESCRIPTION
**Pull request by Sanjays2402**
## 요약

따옴표로 감싼 경로, `file://` URI, 혼합된 Windows 경로 구분자를 기존 워크스페이스 경계 검사 전에 정규화합니다. 경계 검증 자체는 변경하지 않습니다.

## 관련 이슈

Fixes #74

## 변경 종류

- [ ] 버그 수정
- [x] 기능
- [ ] 문서
- [ ] 리팩터링

## 체크리스트

- [x] `python -m pytest -q` 테스트가 통과합니다.
- [ ] `CHANGELOG.md`를 갱신했습니다(해당하는 경우).
- [x] 문서 갱신이 필요한지 확인했습니다.

### CODE REVIEW

File: src/hwpx_mcp_server/workspace.py [hunk 1]
```diff
@@ -306,6 +307,28 @@ def _normalize_roots(values: Iterable[str | os.PathLike[str]]) -> tuple[Path, ..
     return tuple(roots)
 
 
+def _normalize_path_input(
+    value: str | os.PathLike[str], *, windows: bool | None = None
+) -> str:
+    """Normalize common local-path forms before workspace authorization."""
+
+    text = os.fspath(value).strip()
```
Comment (airmang) [inline src/hwpx_mcp_server/workspace.py]: This strips every ordinary path, not just transport decoration. A valid filename with leading/trailing spaces is silently redirected: when both `report.hwpx` and ` report.hwpx ` exist, `resolve(" report.hwpx ")` selects the unspaced file. Output operations can therefore overwrite the wrong document. Please preserve unquoted path text (or reject ambiguous surrounding whitespace) and add a regression test with distinct spaced/unspaced files.

File: src/hwpx_mcp_server/workspace.py [hunk 2]
```diff
@@ -306,6 +307,28 @@ def _normalize_roots(values: Iterable[str | os.PathLike[str]]) -> tuple[Path, ..
...
+    if len(text) >= 2 and text[0] == text[-1] and text[0] in {"'", '"'}:
+        text = text[1:-1].strip()
+
+    parsed = urlsplit(text)
+    if parsed.scheme.lower() == "file":
+        path = unquote(parsed.path)
+        text = f"//{parsed.netloc}{path}" if parsed.netloc else path
```
Comment (airmang) [inline src/hwpx_mcp_server/workspace.py]: `file://localhost/...` denotes a local file, but this turns it into `//localhost/...` (and then a UNC-style path on Windows). An existing file inside the authorized workspace is consequently not found or rejected. Please treat an empty or `localhost` authority as local, handle other authorities explicitly, and add an end-to-end resolver test.

File: src/hwpx_mcp_server/workspace.py [hunk 3]
```diff
@@ -306,6 +307,28 @@ def _normalize_roots(values: Iterable[str | os.PathLike[str]]) -> tuple[Path, ..
...
+def _normalize_path_input(
+    value: str | os.PathLike[str], *, windows: bool | None = None
+) -> str:
+    """Normalize common local-path forms before workspace authorization."""
+
+    text = os.fspath(value).strip()
+    if len(text) >= 2 and text[0] == text[-1] and text[0] in {"'", '"'}:
+        text = text[1:-1].strip()
+
+    parsed = urlsplit(text)
```
Comment (airmang) [inline src/hwpx_mcp_server/workspace.py]: `urlsplit()` can raise `ValueError` for malformed authorities such as `file://[invalid]/inside.hwpx`. That bypasses the typed `WorkspacePathError` contract and reaches callers as an unexpected exception. Please catch URI parser failures and raise `WorkspacePathError(code="WORKSPACE_PATH_INVALID", ...)` without exposing the submitted path; add a regression test for this case.

**Review by airmang — CHANGES_REQUESTED** [review]
Thanks for working on this. The normalization direction makes sense, and the existing test suite plus the basic Ruff gate pass in an isolated checkout. I’m requesting changes because the new pre-authorization normalization currently redirects some valid inputs and leaks an untyped parser error. I reproduced all three cases below. Please preserve exact ordinary path names, handle local file-URI authorities correctly, map malformed URI parsing to the workspace error contract, and add regression tests for each case. Once those are addressed, this should be straightforward to re-review.

**Comment by Sanjays2402** [comment]
Thanks for the detailed repros — all three fixed:

1. **Ordinary paths preserved:** the root cause was a blanket .strip() plus an unconditional urlsplit() on every input. Dropped both — now I only unwrap surrounding quotes, and URI parsing runs *only* when the input actually starts with `file:`. So ` report.hwpx ` (with spaces) stays byte-identical and resolves to the real spaced filename instead of being silently rewritten.

2. **file:// authorities:** empty (`file:///abs`) and `localhost` (`file://localhost/abs`, case-insensitive) authorities now resolve to the plain local path. A non-local authority (`file://remotehost/…`) is rejected with WorkspacePathError(reason="non_local_file_uri_authority") rather than being kept as a `//host` UNC-ish path — let me know if you'd rather it resolve differently, but rejecting felt safer than guessing at a remote mount.

3. **Malformed URIs:** wrapped urlsplit() in try/except so a parser ValueError (e.g. `file://[invalid]/x`) is caught and re-raised as WorkspacePathError(code="WORKSPACE_PATH_INVALID", reason="malformed_file_uri"), and the offending path isn't echoed in the error details.

Added regression tests for each (spaced-name-preserved, localhost/empty authority, non-local rejected, malformed→typed error). Full workspace suite is 45 passed, ruff clean. Pushed.

File: src/hwpx_mcp_server/workspace.py [hunk 4]
```diff
@@ -306,6 +307,63 @@ def _normalize_roots(values: Iterable[str | os.PathLike[str]]) -> tuple[Path, ..
...
+    """
+
+    text = os.fspath(value)
+
+    # Unwrap surrounding quotes only. Quotes are transport decoration, so the
+    # whitespace immediately inside them is decoration too and may be trimmed;
+    # the quoted payload itself is preserved.
+    stripped = text.strip()
+    if len(stripped) >= 2 and stripped[0] == stripped[-1] and stripped[0] in {"'", '"'}:
+        text = stripped[1:-1].strip()
```
Comment (airmang) [inline src/hwpx_mcp_server/workspace.py]: The second `.strip()` still removes meaningful whitespace from the quoted payload. Repro: with both `report.hwpx` and ` report.hwpx ` present, `resolve('" report.hwpx "')` normalizes to `report.hwpx` and selects the unspaced file. This contradicts the docstring above and can still redirect an output operation to the wrong document. The outer `text.strip()` can be used to locate the quote wrapper, but unwrapping should preserve its exact payload (`text = stripped[1:-1]`, without the inner `.strip()`), with a regression test for the quoted spaced filename. If that whitespace is considered ambiguous instead, please reject it with a typed error rather than silently redirecting it.

**Review by airmang — COMMENTED** [review]
Thanks for the update. I verified that the unquoted spaced-name case, localhost/non-local authority handling, and malformed-URI error mapping are now covered, and the branch merges cleanly with the current main test surface. One silent redirection remains for quoted exact path payloads, detailed inline below. The existing changes-requested decision should remain until that case is fixed.

**Comment by airmang** [comment]
Thanks for the quick follow-up. I’m going to take the last quoted-whitespace edge case as a small maintainer fixup now, add the regression test, and validate it against the current `main` suite before merging. Your original commits will be preserved so the contribution remains attributed to you.

File: src/hwpx_mcp_server/workspace.py (hunk 3 above)
Comment (airmang) [reply src/hwpx_mcp_server/workspace.py]: Addressed in Sanjay's `69228bf`: malformed `file:` URIs are mapped to `WorkspacePathError(code="WORKSPACE_PATH_INVALID")` without echoing the submitted path. The regression test and full CI matrix pass on `d07c8ac`.

File: src/hwpx_mcp_server/workspace.py (hunk 1 above)
Comment (airmang) [reply src/hwpx_mcp_server/workspace.py]: Addressed in Sanjay's `69228bf`: unquoted ordinary path text is preserved exactly, and the resolver regression distinguishes spaced and unspaced filenames. Revalidated on head `d07c8ac`; the full Python 3.10/3.12 CI matrix is green.

File: src/hwpx_mcp_server/workspace.py (hunk 2 above)
Comment (airmang) [reply src/hwpx_mcp_server/workspace.py]: Addressed in Sanjay's `69228bf`: empty and `localhost` authorities resolve as local, while non-local authorities fail closed with a typed workspace error. The end-to-end resolver coverage and full CI matrix pass on `d07c8ac`.

File: src/hwpx_mcp_server/workspace.py (hunk 4 above)
Comment (airmang) [reply src/hwpx_mcp_server/workspace.py]: Addressed in maintainer fixup `d07c8ac`: quote detection still uses the outer stripped view, but unwrapping now preserves the exact payload with `text = stripped[1:-1]`. Added both normalization and resolver regressions for a quoted spaced filename; public hygiene and the full Python 3.10/3.12 CI matrix are green.

**Review by airmang — APPROVED** [review]
All requested path-normalization cases are resolved with regression coverage: exact ordinary and quoted whitespace preservation, local/non-local file URI authority handling, and typed malformed-URI errors. The maintainer fixup was limited to the final quoted-whitespace edge case and public-hygiene-safe test fixtures. The PR merges cleanly with current `main`; Tests (Python 3.10/3.12 plus clean installs), CodeQL, and Dependency review all pass.

**Comment by airmang** [comment]
Thanks again for the contribution — merged via rebase. Your two authored commits are preserved on `main`, and GitHub now lists `Sanjays2402` in the repository contributors with two contributions. The linked issue #74 closed automatically.
